### ISSUE TITLE
CV01 rule fails to detect not-equal operators split across lines or with comments

### ISSUE THREAD

**Issue by Fank** [quarylabs/sqruff#1872]
# CV01 Rule Bug: Fails to Detect Multi-Line Not-Equal Operators

## Summary
The CV01 rule (`convention.not_equal`) fails to detect and convert not-equal operators (`<>` ↔ `!=`) when they are split across lines or have comments between the operator tokens. This causes test failures and prevents the rule from working on formatted SQL code.

## Failing Test Case
The failing test `test_fail_c_style_not_equal_to_tsql` demonstrates this issue:

**Input (ANSI style):**
```sql
SELECT * FROM X WHERE 1  <
  -- some comment
> 2
```

**Expected Output (C-style):**
```sql
SELECT * FROM X WHERE 1  !
  -- some comment
= 2
```

**Actual Result:** No change applied (rule doesn't detect the pattern)

## Root Cause Analysis

The CV01 rule implementation in `crates/lib/src/rules/convention/cv01.rs` has logic that expects the `<` and `>` (or `!` and `=`) tokens to be direct sequential children of a comparison operator node:

```rust
let raw_operator_list = raw_comparison_operators
    .iter()
    .map(|r| r.raw())
    .collect::<Vec<_>>();
if raw_operator_list != ["<", ">"] && raw_operator_list != ["!", "="] {
    return Vec::new();
}
```

When comments or newlines appear between the operator tokens, the AST structure changes and this pattern matching fails.

## Comprehensive Test Results

I tested 10 different scenarios with both `c_style` and `ansi` preferences:

### ✅ **WORKING CASES (4 out of 10):**
| Test Case | Input | Expected Behavior | Result |
|-----------|-------|-------------------|--------|
| Simple with spaces | `1 <> 2` | Convert to `1 != 2` | ✅ Works |
| Simple without spaces | `1<>2` | Convert to `1!=2` | ✅ Works |
| Simple with spaces | `1 != 2` | Convert to `1 <> 2` | ✅ Works |
| Simple without spaces | `1!=2` | Convert to `1<>2` | ✅ Works |

### ❌ **FAILING CASES (6 out of 10):**
| Test Case | Input | Expected Behavior | Result |
|-----------|-------|-------------------|--------|
| Line comment between | `1 <\n-- comment\n> 2` | Convert to `1 !\n-- comment\n= 2` | ❌ No change |
| Block comment between | `1 < /* comment */ > 2` | Convert to `1 ! /* comment */ = 2` | ❌ No change |
| Simple newline split | `1 <\n> 2` | Convert to `1 !\n= 2` | ❌ No change |
| C-style with line comment | `1 !\n-- comment\n= 2` | Convert to `1 <\n-- comment\n> 2` | ❌ No change |
| C-style with block comment | `1 ! /* comment */ = 2` | Convert to `1 < /* comment */ > 2` | ❌ No change |
| C-style with newline | `1 !\n= 2` | Convert to `1 <\n> 2` | ❌ No change |

## Impact
- **Test failures**: Causes test suite failures as demonstrated
- **Real-world usage**: Many SQL formatters and developers split operators across lines for readability
- **Inconsistent behavior**: Rule works inconsistently depending on formatting

## Environment
- **Version**: Current main branch (as of 2025-01-08)
- **Dialect**: T-SQL (though likely affects other dialects too)
- **Rule**: CV01 (`convention.not_equal`)

## Reproduction Steps
1. Create SQL file with multi-line not-equal operator:
   ```sql
   SELECT * FROM X WHERE 1  <
     -- some comment
   > 2
   ```
2. Configure sqruff with CV01 rule and `c_style` preference
3. Run `sqruff fix` on the file
4. Observe that no changes are applied

## Expected Fix
The CV01 rule should be enhanced to:
1. Recognize comparison operators even when split across lines
2. Handle comments (both line and block) between operator tokens
3. Preserve the original formatting structure when applying fixes
4. Work consistently regardless of whitespace or comment placement

## Technical Details
The issue is in the `eval` method's assumption that comparison operator children will be exactly `["<", ">"]` or `["!", "="]`. The rule needs to be modified to:
1. Walk the AST more intelligently to find the actual operator tokens
2. Handle cases where whitespace and comments separate the tokens
3. Apply fixes that maintain the original structure while only changing the operator characters

This affects the test case `test_fail_c_style_not_equal_to_tsql` in `crates/lib/test/fixtures/rules/std_rule_cases/CV01.yml`.
